crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -573,6 +576,38 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
             payment_method_info,
         } = mandate_details;
 
+        let additional_pm_data_from_locker = if let Some(ref pm) = payment_method_info {
+            let card_detail_from_locker: Option<api::CardDetailFromLocker> = pm
+                .payment_method_data
+                .clone()
+                .map(|x| x.into_inner().expose())
+                .and_then(|v| match serde_json::from_value::<PaymentMethodsData>(v) {
+                    Ok(data) => Some(data),
+                    Err(err) => {
+                        router_env::logger::info!(
+                            "PaymentMethodsData deserialization failed : {err}"
+                        );
+                        None
+                    }
+                })
```

Comment:
Can use `parse_value()` here instead, you can then log the error and call `.ok()`.

Also, are we okay with proceeding further if we can't deserialize payment method from the locker?

Reply:
@Aprabhat19 Can you verify if we can pass None to the `additional_payment_method` if deserialization fails?

Reply:
if deserialization fails ,we can proceed further, its just additional pm data